drivers/sensor/memsic/mmc56x3/mmc56x3.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_MMC56X3_MMC56X3_H_
+#define ZEPHYR_DRIVERS_SENSOR_MMC56X3_MMC56X3_H_
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/rtio/rtio.h>
+#include <zephyr/drivers/sensor/mmc56x3.h>
+
+#define DT_DRV_COMPAT memsic_mmc56x3
+
+#define MMC56X3_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+union mmc56x3_bus {
+	struct i2c_dt_spec i2c;
+};
+
+typedef int (*mmc56x3_bus_check_fn)(const union mmc56x3_bus *bus);
+typedef int (*mmc56x3_reg_read_fn)(const union mmc56x3_bus *bus,
+				  uint8_t start, uint8_t *buf, int size);
+typedef int (*mmc56x3_reg_write_fn)(const union mmc56x3_bus *bus,
+				   uint8_t reg, uint8_t val);
+typedef int (*mmc56x3_raw_read_fn)(const union mmc56x3_bus *bus,
+				  uint8_t *buf, size_t size);
+typedef int (*mmc56x3_raw_write_fn)(const union mmc56x3_bus *bus,
+				  uint8_t *buf, size_t size);
+
+struct mmc56x3_bus_io {
+	mmc56x3_bus_check_fn check;
+	mmc56x3_reg_read_fn read;
+	mmc56x3_reg_write_fn write;
+	mmc56x3_raw_read_fn raw_read;
+	mmc56x3_raw_write_fn raw_write;
+};
+
+extern const struct mmc56x3_bus_io mmc56x3_bus_io_i2c;
+
+#define MMC56X3_REG_TEMP                   0x09
+#define MMC56X3_CHIP_ID                    0x10
+#define MMC56X3_REG_STATUS                 0x18
+#define MMC56X3_REG_INTERNAL_ODR           0x1a
+#define MMC56X3_REG_INTERNAL_CTRL_0        0x1b
+#define MMC56X3_REG_INTERNAL_CTRL_1        0x1c
+#define MMC56X3_REG_INTERNAL_CTRL_2        0x1d
+#define MMC56X3_REG_ID                     0x39
+
+#define MMC56X3_CMD_RESET                  0x10
+#define MMC56X3_CMD_SET                    0x08
+#define MMC56X3_CMD_SW_RESET               0x80
+#define MMC56X3_CMD_TAKE_MEAS_M            0x01
+#define MMC56X3_CMD_TAKE_MEAS_T            0x02
+#define MMC56X3_CMD_AUTO_SELF_RESET_EN     0x20
+#define MMC56X3_CMD_CMM_FREQ_EN            0x80
+#define MMC56X3_CMD_CMM_EN                 0x10
+#define MMC56X3_CMD_HPOWER                 0x80
+
+#define MMC56X3_STATUS_MEAS_M_DONE         0x80
+#define MMC56X3_STATUS_MEAS_T_DONE         0x40
+
+#define MMC56X3_REG_MAGN_X_OUT_0           0x00
+/* Range is -30 to 30, sensitivity of raw 20-bit reading is
+ * 16384 = 1 Gauss. To convert raw reading to
+ * Q5.26 with range -32 to 32,
+ * reading * (1/16384) * pow(2, 31)/32
+ * reading * 4096
+ */
+#define MMC56X3_MAGN_CONV_RAW_Q5_26_20B    4096
+/* 1/16384 */
+#define MMC56X3_MAGN_GAUSS_RES             0.000061035
```

Comment:
I don't believe this is used anymore is it?

Reply:
If you mean MMC56X3_MAGN_GAUSS_RES, it's used in mmc56x3_channel_get:

```
static int mmc56x3_channel_get(const struct device *dev,
			      enum sensor_channel chan,
			      struct sensor_value *val)
{
	struct mmc56x3_data *data = dev->data;

	switch (chan) {
	case SENSOR_CHAN_AMBIENT_TEMP:
		convert_double_to_sensor_val(
			MMC56X3_TEMP_BASE + (data->temp * MMC56X3_TEMP_RES), val);
		break;
	case SENSOR_CHAN_MAGN_X:
		convert_double_to_sensor_val(
			data->magn_x * MMC56X3_MAGN_GAUSS_RES, val);
		break;
	case SENSOR_CHAN_MAGN_Y:
		convert_double_to_sensor_val(
			data->magn_y * MMC56X3_MAGN_GAUSS_RES, val);
		break;
	case SENSOR_CHAN_MAGN_Z:
		convert_double_to_sensor_val(
			data->magn_z * MMC56X3_MAGN_GAUSS_RES, val);
		break;
	case SENSOR_CHAN_MAGN_XYZ:
		convert_double_to_sensor_val(
			data->magn_x * MMC56X3_MAGN_GAUSS_RES, val);
		convert_double_to_sensor_val(
			data->magn_y * MMC56X3_MAGN_GAUSS_RES, val + 1);
		convert_double_to_sensor_val(
			data->magn_z * MMC56X3_MAGN_GAUSS_RES, val + 2);
		break;
	default:
		return -ENOTSUP;
	}

	return 0;
}
```

Reply:
Ah right, makes sense now